src/crimson/os/seastore/cache.h

```diff
@@ -449,6 +449,7 @@ class Cache {
     Transaction &t,
     CachedExtentRef extent)
   {
+    assert(extent->is_stable());
```

Comment:
Looks is_initial_pending() is also possible here.

Reply:
Seems all kinds of valid extents are possible, parent/child link does apply to mutation-pending extents (https://github.com/ceph/ceph/pull/55831#issuecomment-1985137599)

Reply:
Changed to is_valid() only.

Reply:
Looking at the callers of `get_extent_viewable_by_trans()`, seems to me that the `extent` can be mutation-pending only if it is root, so the invariant here could be:
`assert(extent->is_root() ? extent->is_valid() : (extent->is_stable() || extent->is_initial_pending()));`